genai-use-cases/aws-glue-metadata-generation/how_to_generate_metadata_for_glue_data_catalog_w_bedrock.ipynb

```diff
@@ -0,0 +1,846 @@
+    "5. Remove the DatabaseName, CreatedBy, IsRegisteredWithLakeFormation, CatalogId,VersionId,IsMultiDialectView,CreateTime, UpdateTime. \n",
+    "6. Write the table description in the Description attribute. Ensure you use information from the <documentation> if possible.\n",
+    "7. List all the table columns under the attribute \"StorageDescriptor\" and then the attribute Columns. Add Location, InputFormat, and SerdeInfo\n",
+    "8. For each column in the StorageDescriptor, add the attribute \"Comment\". Ensure Name and Type are included.  If a table uses a composite primary key, then the order of a given column in a table’s primary key is listed in parentheses following the column name.\n",
+    "9. Your response must be a valid JSON object.\n",
+    "10. Ensure that the data is accurately represented and properly formatted within the JSON structure. The resulting JSON table should provide a clear, structured overview of the information presented in the original text.\n",
+    "11. If  you cannot think of an accurate description of a column, say 'not available'\n",
+    "\n",
+    "<glue_data_catalog>\n",
+    "{data_catalog}\n",
+    "</glue_data_catalog>\n",
+    "\n",
+    "Here is some additional information about the database in <notes></notes> tags.\n",
+    "<notes>\n",
+    "Typically foreign key columns consist of the name of the table plus the id suffix\n",
+    "<notes>\n",
+    "\"\"\"\n",
+    "\n",
+    "\n",
+    "messages = [\n",
+    "    (\"system\", system),\n",
+    "    (\"user\", user_msg_rag),\n",
+    "]\n",
+    "\n",
+    "prompt = ChatPromptTemplate.from_messages(messages)\n",
+    "\n",
+    "# Retrieve and Generate\n",
+    "retriever = vs.as_retriever(\n",
+    "    search_type=\"similarity\",\n",
+    "    search_kwargs={\"k\": 3},\n",
+    ")\n",
+    "\n",
+    "chain = (   \n",
+    "    {\"context\": itemgetter(\"table\")| retriever, \"data_catalog\": itemgetter(\"data_catalog\"), \"table\": itemgetter(\"table\")}\n",
+    "    | prompt\n",
+    "    | model\n",
+    "    | StrOutputParser()\n",
+    ")\n",
+    "\n",
+    "TableInputFromLLM = chain.invoke({\"data_catalog\":glue_data_catalog, \"table\":table})\n",
+    "print(TableInputFromLLM)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9e7ce232-0411-41b5-abb6-71141c467e8c",
+   "metadata": {},
+   "source": [
+    "### Update the AWS Glue Data Catalog"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c9e8b43e-d560-4fb5-8613-5e31c5260ef1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "validate(instance=json.loads(TableInputFromLLM), schema=schema_table_input)\n",
+    "\n",
+    "response = glue_client.update_table(DatabaseName=database, TableInput= json.loads(TableInputFromLLM) )\n",
+    "print(f\"Table {table} metadata updated!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cda324f7-1664-4ce9-be0e-50b31b5bb193",
+   "metadata": {},
+   "source": [
+    "# Clean Up"
```

Comment:
Maybe add remove the glue crawler role

Reply:
added instructions, however as we assume the Crawler is created outside the notebook, this is an optional step.